File: strings/02_target_n_PushPop_stack/target_n_PushPop_stack.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
#include <map>
#include <cassert>

using namespace std;


class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
        assert(n >= 1 && n <= 100);
        assert(target.size() >= 1 && target.size() <= 100);
        assert(is_sorted(target.begin(), target.end()));

        vector<string> stack;
        int target_cnt = 0;

        for (int stream_num = 1; stream_num <= n; ++stream_num) {
            stack.push_back("Push");
            if (stream_num == target[target_cnt]) {
                target_cnt++;
                if (target_cnt == target.size()) {
                    return stack;
                }
            } else {
                stack.push_back("Pop");
            }
        }

        return stack;
    }
};

File: strings/02_target_n_PushPop_stack/target_n_PushPop_stack.py
from typing import List


class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:
        assert n >= 1 and n <= 100
        assert len(target) >= 1 and len(target) <= 100
        assert all(target[i] <= target[i + 1] for i in range(len(target) - 1))

        stack: List[str] = []
        target_cnt = 0

        for stream_num in range(1, n + 1):
            stack.append("Push")
            if stream_num == target[target_cnt]:
                target_cnt += 1
                if target_cnt == len(target):
                    return stack
            else:
                stack.append("Pop")

        return stack
